sdk/servicebus/Azure.Messaging.ServiceBus/tests/Receiver/SessionReceiverLiveTests.cs

```diff
@@ -1144,5 +1144,55 @@ await using (var scope = await ServiceBusScope.CreateWithQueue(enablePartitionin
                         .EqualTo(ServiceBusFailureReason.SessionLockLost));
             }
         }
+
+        [Test]
+        public async Task SessionOrderingIsGuaranteed()
+        {
+            await using (var scope = await ServiceBusScope.CreateWithQueue(enablePartitioning: false, enableSession: true))
```

Comment:
Just curious does this fail without the change? I feel like I remember the ordering issue being kind of random.

Reply:
Yes it does! I think the key is to make sure the receive calls don't get all the messages that they requested and to have new messages sent to the queue periodically. When I tested it out without the change it failed every time.

Reply:
Cool, that makes sense